File: modulo-01-fundamentos-arquitectura/codigo/src/main/java/pe/joedayz/microservicios/modulo01/ddd/order/OrderId.java
package pe.joedayz.microservicios.modulo01.ddd.order;

import java.util.UUID;

/**
 * Identidad de la entidad Order. A diferencia de un Value Object, define IDENTIDAD:
 * dos pedidos con los mismos datos pero distinto OrderId son pedidos distintos.
 */
public record OrderId(UUID value) {

    public static OrderId newId() {
        return new OrderId(UUID.randomUUID());
    }

    public static OrderId of(String value) {
        return new OrderId(UUID.fromString(value));
    }

    @Override
    public String toString() {
        return value.toString();
    }
}

File: modulo-01-fundamentos-arquitectura/codigo-dotnet/src/Modulo01.Fundamentos/Ddd/Orders/OrderId.cs
namespace JoeDayz.Microservicios.Modulo01.Ddd.Orders;

/// <summary>Identidad del agregado Pedido.</summary>
public readonly record struct OrderId(Guid Value)
{
    public static OrderId New() => new(Guid.NewGuid());

    public static OrderId Of(string value) => new(Guid.Parse(value));

    public override string ToString() => Value.ToString();
}
